packages/proxy/src/proxy.ts

```diff
@@ -636,14 +637,27 @@ export async function proxyV1({
 
           try {
             if ("data" in event) {
-              const result = JSON.parse(event.data) as ChatCompletionChunk;
+              const result = JSON.parse(event.data) as
+                | ChatCompletionChunk
+                | undefined;
               if (result) {
-                if (result.usage) {
+                const extendedUsage = completionUsageSchema.safeParse(
+                  result.usage,
+                );
+                if (extendedUsage.success) {
                   spanLogger.log({
                     metrics: {
-                      tokens: result.usage.total_tokens,
-                      prompt_tokens: result.usage.prompt_tokens,
-                      completion_tokens: result.usage.completion_tokens,
+                      tokens: extendedUsage.data.total_tokens,
+                      prompt_tokens: extendedUsage.data.prompt_tokens,
+                      completion_tokens: extendedUsage.data.completion_tokens,
+                      prompt_cached_tokens:
+                        extendedUsage.data.prompt_tokens_details?.cached_tokens,
+                      prompt_cache_creation_tokens:
+                        extendedUsage.data.prompt_tokens_details
+                          ?.cache_creation_tokens,
+                      completion_reasoning_tokens:
```

Comment:
will need to add this to the STANDARD_METRICS: https://github.com/braintrustdata/braintrust/pull/5540/files. we should see if the  detection is working, and would have caught this